First iteration

Zarecs opened the pull request.

Zarecs committed:
First

Zarecs committed:
fix import

Zarecs committed:
fix daysteps

Zarecs committed:
fix log problem

dirtymew commented on internal/daysteps/daysteps.go:
```diff
@@ -1,20 +1,104 @@
+	parts := strings.Split(data, ",")
+	if len(parts) != 2 {
+		log.Println("сообщение")
+		return 0, 0, fmt.Errorf("неверный формат данных")
```
ошибки лучше писать на английском

Zarecs replied:
Перевел ошибки на английский, кроме мест где требуется русский текст для тестов.
Код из теста:
for _, tt := range tests {
		suite.Run(tt.name, func() {
			got, err := TrainingInfo(tt.input, tt.weight, tt.height)

			if tt.wantErr {
				assert.Error(suite.T(), err)
				assert.Empty(suite.T(), got)
				if tt.name == "неизвестный тип тренировки - проверка текста ошибки" {
					assert.Contains(suite.T(), err.Error(), "неизвестный тип тренировки")
				}
				return
			}

			assert.NoError(suite.T(), err)
			assert.Equal(suite.T(), tt.want, got)
		})

dirtymew replied:
это требование задания - еще не успели поправить контент (

dirtymew commented on internal/daysteps/daysteps.go:
```diff
@@ -1,20 +1,104 @@
-	// TODO: реализовать функцию
+	parts := strings.Split(data, ",")
+	if len(parts) != 2 {
+		log.Println("сообщение")
```
сообщения в логе должны быть информативными

dirtymew replied:
везде исправить или убрать

Zarecs replied:
Не получается прикрепить сюда скрин,  отвечал через почту.  вот текст комментария:
"Константин Селезнев [наставник]

12:00
Привет! Да, какое-то серьезное расхождение задания с ожиданиями в тестах... Будем разбираться



Чтобы сейчас не блокироваться, рекомендую в daysteps.DayActionInfo добавить логирование всех ошибок. Нужно просто в соответствующих местах добавить строчку:





log.Println("сообщение")
"

dirtymew replied:
ты слишком его буквально понял - для разных случаев текст лога будет разный )

Zarecs replied:
ну как я понял, он предложил поставить заглушку, так как в тексте задания о логах нет ни слова. 
По этому и взял предложенный текст, но согласен с точки зрения навыков, стоит выдавать более информативные данные в лог.

dirtymew commented on internal/daysteps/daysteps.go:
```diff
@@ -1,20 +1,104 @@
+	}
+
+	// Обработка знака "+"
+	if strings.HasPrefix(stepsStr, "+") {
```
зачем эта проверка?

dirtymew replied:
лишнее

Zarecs replied:
Удалил данную проверку.

dirtymew commented on internal/daysteps/daysteps.go:
```diff
@@ -1,20 +1,104 @@
+	}
+
+	// Проверка на наличие только знака + или -
+	if stepsStr == "+" || stepsStr == "-" {
```
лишнее - не нужно

Zarecs replied:
Удалил данную проверку.

dirtymew commented on internal/daysteps/daysteps.go:
```diff
@@ -1,20 +1,104 @@
+	}
+	// проверка на пробелы
+	trimmedStepsStr := strings.TrimSpace(stepsStr)
+	if trimmedStepsStr != stepsStr {
```
тоже лишнее

Zarecs replied:
Удалил данную проверку.

dirtymew commented on internal/daysteps/daysteps.go:
```diff
@@ -1,20 +1,104 @@
+	num, err := strconv.Atoi(stepsStr)
+	if err != nil {
+		log.Println("сообщение")
+		return 0, 0, fmt.Errorf("ошибка парсинга количества шагов: %v", err)
```
ошибка неверно обернута - нужно через %w

dirtymew replied:
везде исправить

Zarecs replied:
Обновил обертку ошибки на %w

dirtymew commented on internal/daysteps/daysteps.go:
```diff
@@ -1,20 +1,104 @@
+	}
+
+	// Проверка на пробелы внутри времени
+	if strings.Contains(timeStr, " ") {
```
лишняя проверка

Zarecs replied:
Проверка сделана для прохождения теста
тест:
             {
			name:         "неверная продолжительность - пробел между числом и единицей",
			input:        "678,1 h30m",
			wantSteps:    0,
			wantDuration: 0,
			wantErr:      true,
		},

dirtymew commented on internal/daysteps/daysteps.go:
```diff
@@ -1,20 +1,104 @@
+
+	// Обработка времени
+	timeStr := parts[1]
+	if timeStr == "" {
```
лишнее - лучше проверять значение после ParseDuration

Zarecs replied:
Удалил данную проверку

dirtymew commented on internal/spentcalories/spentcalories.go:
```diff
@@ -13,26 +16,103 @@ const (
 }
 
-func meanSpeed(steps int, height float64, duration time.Duration) float64 {
+func meanSpeed(steps int, height float64, duration time.Duration) float64 { //ok
```
комментарии тоже должны быть информативные

Zarecs replied:
удалил комментарии "ОК" у функций.

dirtymew commented on internal/spentcalories/spentcalories.go:
```diff
@@ -13,26 +16,103 @@ const (
+	switch activityType {
+	case "Ходьба":
+		calories, _ = WalkingSpentCalories(steps, weight, height, duration)
+		activityName = "Ходьба"
```
это лишнее

dirtymew replied:
есть же activityType

Zarecs replied:
в последней итерации обновил данный блок.

dirtymew commented on internal/spentcalories/spentcalories.go:
```diff
@@ -13,26 +16,103 @@ const (
+		activityName = "Ходьба"
+	case "Бег":
+		calories, _ = RunningSpentCalories(steps, weight, height, duration)
+		activityName = "Бег"
```
это лишнее

Zarecs replied:
удалил activityName

dirtymew commented on internal/spentcalories/spentcalories.go:
```diff
@@ -13,26 +16,103 @@ const (
+func RunningSpentCalories(steps int, weight, height float64, duration time.Duration) (float64, error) { // ok
 	// TODO: реализовать функцию
+	if steps <= 0 || weight <= 0 || height <= 0 || duration <= 0 {
+		return 0, fmt.Errorf("неверный формат данных")
```
ошибка не информативная - по ней не понятно что именно не так

Zarecs replied:
Обновил текст ошибки

dirtymew reviewed (changes requested):
Привет Владислав

по работе есть замечания - нужно поправить

Zarecs commented:
Добрый день, Антон.
Спасибо за фидбек, касательно ошибки в лог, сделал как рекомендовал
наставник. Скриншот прикрепляю.

По остальным ошибкам выпущу обновление.

Спасибо.

Best regards
Vladislav Nekrasov
Tel: +79601772423

вс, 18 мая 2025 г., 22:30 Anton Sattarov ***@***.***>:

> ***@***.**** requested changes on this pull request.
>
> Привет Владислав
>
> по работе есть замечания - нужно поправить
> ------------------------------
>
> In internal/daysteps/daysteps.go
> <https://github.com/Zarecs/spin1/pull/1#discussion_r2094606662>:
>
> >  )
>
>  func parsePackage(data string) (int, time.Duration, error) {
> -	// TODO: реализовать функцию
> +	parts := strings.Split(data, ",")
> +	if len(parts) != 2 {
> +		log.Println("сообщение")
> +		return 0, 0, fmt.Errorf("неверный формат данных")
>
> ошибки лучше писать на английском
> ------------------------------
>
> In internal/daysteps/daysteps.go
> <https://github.com/Zarecs/spin1/pull/1#discussion_r2094606709>:
>
> >  )
>
>  func parsePackage(data string) (int, time.Duration, error) {
> -	// TODO: реализовать функцию
> +	parts := strings.Split(data, ",")
> +	if len(parts) != 2 {
> +		log.Println("сообщение")
>
> сообщения в логе должны быть информативными
> ------------------------------
>
> In internal/daysteps/daysteps.go
> <https://github.com/Zarecs/spin1/pull/1#discussion_r2094606756>:
>
> > +
> +	// Обработка количества шагов
> +	stepsStr := parts[0]
> +	if stepsStr == "" {
> +		log.Println("сообщение")
> +		return 0, 0, fmt.Errorf("количество шагов не указано")
> +	}
> +
> +	// Проверка на наличие только знака + или -
> +	if stepsStr == "+" || stepsStr == "-" {
> +		log.Println("сообщение")
> +		return 0, 0, fmt.Errorf("неверный формат количества шагов")
> +	}
> +
> +	// Обработка знака "+"
> +	if strings.HasPrefix(stepsStr, "+") {
>
> зачем эта проверка?
> ------------------------------
>
> In internal/daysteps/daysteps.go
> <https://github.com/Zarecs/spin1/pull/1#discussion_r2094606846>:
>
> > -	// TODO: реализовать функцию
> +	parts := strings.Split(data, ",")
> +	if len(parts) != 2 {
> +		log.Println("сообщение")
> +		return 0, 0, fmt.Errorf("неверный формат данных")
> +	}
> +
> +	// Обработка количества шагов
> +	stepsStr := parts[0]
> +	if stepsStr == "" {
> +		log.Println("сообщение")
> +		return 0, 0, fmt.Errorf("количество шагов не указано")
> +	}
> +
> +	// Проверка на наличие только знака + или -
> +	if stepsStr == "+" || stepsStr == "-" {
>
> лишнее - не нужно
> ------------------------------
>
> In internal/daysteps/daysteps.go
> <https://github.com/Zarecs/spin1/pull/1#discussion_r2094606946>:
>
> > +		return 0, 0, fmt.Errorf("количество шагов не указано")
> +	}
> +
> +	// Проверка на наличие только знака + или -
> +	if stepsStr == "+" || stepsStr == "-" {
> +		log.Println("сообщение")
> +		return 0, 0, fmt.Errorf("неверный формат количества шагов")
> +	}
> +
> +	// Обработка знака "+"
> +	if strings.HasPrefix(stepsStr, "+") {
> +		stepsStr = stepsStr[1:]
> +	}
> +	// проверка на пробелы
> +	trimmedStepsStr := strings.TrimSpace(stepsStr)
> +	if trimmedStepsStr != stepsStr {
>
> тоже лишнее
> ------------------------------
>
> In internal/daysteps/daysteps.go
> <https://github.com/Zarecs/spin1/pull/1#discussion_r2094607029>:
>
> > +
> +	// Обработка знака "+"
> +	if strings.HasPrefix(stepsStr, "+") {
> +		stepsStr = stepsStr[1:]
> +	}
> +	// проверка на пробелы
> +	trimmedStepsStr := strings.TrimSpace(stepsStr)
> +	if trimmedStepsStr != stepsStr {
> +		log.Println("сообщение")
> +		return 0, 0, fmt.Errorf("количество шагов не должно содержать пробелов в начале или конце")
> +	}
> +
> +	num, err := strconv.Atoi(stepsStr)
> +	if err != nil {
> +		log.Println("сообщение")
> +		return 0, 0, fmt.Errorf("ошибка парсинга количества шагов: %v", err)
>
> ошибка неверно обернута - нужно через %w
> ------------------------------
>
> In internal/daysteps/daysteps.go
> <https://github.com/Zarecs/spin1/pull/1#discussion_r2094607166>:
>
> > +		return 0, 0, fmt.Errorf("ошибка парсинга количества шагов: %v", err)
> +	}
> +	if num <= 0 {
> +		log.Println("сообщение")
> +		return 0, 0, fmt.Errorf("количество шагов должно быть положительным")
> +	}
> +
> +	// Обработка времени
> +	timeStr := parts[1]
> +	if timeStr == "" {
> +		log.Println("сообщение")
> +		return 0, 0, fmt.Errorf("время тренировки не указано")
> +	}
> +
> +	// Проверка на пробелы внутри времени
> +	if strings.Contains(timeStr, " ") {
>
> лишняя проверка
> ------------------------------
>
> In internal/daysteps/daysteps.go
> <https://github.com/Zarecs/spin1/pull/1#discussion_r2094607287>:
>
> > +		return 0, 0, fmt.Errorf("количество шагов не должно содержать пробелов в начале или конце")
> +	}
> +
> +	num, err := strconv.Atoi(stepsStr)
> +	if err != nil {
> +		log.Println("сообщение")
> +		return 0, 0, fmt.Errorf("ошибка парсинга количества шагов: %v", err)
> +	}
> +	if num <= 0 {
> +		log.Println("сообщение")
> +		return 0, 0, fmt.Errorf("количество шагов должно быть положительным")
> +	}
> +
> +	// Обработка времени
> +	timeStr := parts[1]
> +	if timeStr == "" {
>
> лишнее - лучше проверять значение после ParseDuration
> ------------------------------
>
> In internal/spentcalories/spentcalories.go
> <https://github.com/Zarecs/spin1/pull/1#discussion_r2094607452>:
>
> >  }
>
> -func meanSpeed(steps int, height float64, duration time.Duration) float64 {
> +func meanSpeed(steps int, height float64, duration time.Duration) float64 { //ok
>
> комментарии тоже должны быть информативные
> ------------------------------
>
> In internal/spentcalories/spentcalories.go
> <https://github.com/Zarecs/spin1/pull/1#discussion_r2094607547>:
>
> >
> -func RunningSpentCalories(steps int, weight, height float64, duration time.Duration) (float64, error) {
> +	var calories float64
> +	var activityName string
> +
> +	switch activityType {
> +	case "Ходьба":
> +		calories, _ = WalkingSpentCalories(steps, weight, height, duration)
> +		activityName = "Ходьба"
>
> это лишнее
> ------------------------------
>
> In internal/spentcalories/spentcalories.go
> <https://github.com/Zarecs/spin1/pull/1#discussion_r2094607582>:
>
> >
> -func RunningSpentCalories(steps int, weight, height float64, duration time.Duration) (float64, error) {
> +	var calories float64
> +	var activityName string
> +
> +	switch activityType {
> +	case "Ходьба":
> +		calories, _ = WalkingSpentCalories(steps, weight, height, duration)
> +		activityName = "Ходьба"
> +	case "Бег":
> +		calories, _ = RunningSpentCalories(steps, weight, height, duration)
> +		activityName = "Бег"
>
> это лишнее
> ------------------------------
>
> In internal/spentcalories/spentcalories.go
> <https://github.com/Zarecs/spin1/pull/1#discussion_r2094607673>:
>
> >  	// TODO: реализовать функцию
> +	if steps <= 0 || weight <= 0 || height <= 0 || duration <= 0 {
> +		return 0, fmt.Errorf("неверный формат данных")
>
> ошибка не информативная - по ней не понятно что именно не так
> ------------------------------
>
> In internal/daysteps/daysteps.go
> <https://github.com/Zarecs/spin1/pull/1#discussion_r2094607103>:
>
> >  )
>
>  func parsePackage(data string) (int, time.Duration, error) {
> -	// TODO: реализовать функцию
> +	parts := strings.Split(data, ",")
> +	if len(parts) != 2 {
> +		log.Println("сообщение")
>
> везде исправить или убрать
> ------------------------------
>
> In internal/daysteps/daysteps.go
> <https://github.com/Zarecs/spin1/pull/1#discussion_r2094606784>:
>
> > +
> +	// Обработка количества шагов
> +	stepsStr := parts[0]
> +	if stepsStr == "" {
> +		log.Println("сообщение")
> +		return 0, 0, fmt.Errorf("количество шагов не указано")
> +	}
> +
> +	// Проверка на наличие только знака + или -
> +	if stepsStr == "+" || stepsStr == "-" {
> +		log.Println("сообщение")
> +		return 0, 0, fmt.Errorf("неверный формат количества шагов")
> +	}
> +
> +	// Обработка знака "+"
> +	if strings.HasPrefix(stepsStr, "+") {
>
> лишнее
> ------------------------------
>
> In internal/daysteps/daysteps.go
> <https://github.com/Zarecs/spin1/pull/1#discussion_r2094607079>:
>
> > +
> +	// Обработка знака "+"
> +	if strings.HasPrefix(stepsStr, "+") {
> +		stepsStr = stepsStr[1:]
> +	}
> +	// проверка на пробелы
> +	trimmedStepsStr := strings.TrimSpace(stepsStr)
> +	if trimmedStepsStr != stepsStr {
> +		log.Println("сообщение")
> +		return 0, 0, fmt.Errorf("количество шагов не должно содержать пробелов в начале или конце")
> +	}
> +
> +	num, err := strconv.Atoi(stepsStr)
> +	if err != nil {
> +		log.Println("сообщение")
> +		return 0, 0, fmt.Errorf("ошибка парсинга количества шагов: %v", err)
>
> везде исправить
> ------------------------------
>
> In internal/spentcalories/spentcalories.go
> <https://github.com/Zarecs/spin1/pull/1#discussion_r2094607605>:
>
> >
> -func RunningSpentCalories(steps int, weight, height float64, duration time.Duration) (float64, error) {
> +	var calories float64
> +	var activityName string
> +
> +	switch activityType {
> +	case "Ходьба":
> +		calories, _ = WalkingSpentCalories(steps, weight, height, duration)
> +		activityName = "Ходьба"
>
> есть же activityType
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Zarecs/spin1/pull/1#pullrequestreview-2849070408>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BRDPK4LB3KMQCOPV3KLTHML27DNVTAVCNFSM6AAAAAB5K2LH6KVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDQNBZGA3TANBQHA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

Zarecs committed:
Correction of comments

Zarecs committed:
correction of comments v2

dirtymew commented on internal/spentcalories/spentcalories.go:
```diff
@@ -15,24 +18,105 @@ const (
+	var calories float64
 
-func RunningSpentCalories(steps int, weight, height float64, duration time.Duration) (float64, error) {
+	if activityType == "Ходьба" {
```
тут лучше подойдет switch

Zarecs replied:
Обновил данный блок на switch

dirtymew commented on internal/spentcalories/spentcalories.go:
```diff
@@ -15,24 +18,105 @@ const (
-func RunningSpentCalories(steps int, weight, height float64, duration time.Duration) (float64, error) {
+	if activityType == "Ходьба" {
+		calories, _ = WalkingSpentCalories(steps, weight, height, duration)
+	} else if activityType == "Бег" {
```
else if никогда не используй

else только если без него не обойтись и читаемость кода не страдает

Zarecs replied:
Обновил данный блок на switch

dirtymew commented:
@Zarecs скриншота нет в комментарии - ничего не могу сказать поэтому поводу

и как что сделано и к какому замечанию это относится я не понял

на github можно удобно по месту кода в PR оставлять комментарии

если еще актуально сделай так и приложи скриншоты - тогда смогу ответить

dirtymew reviewed (changes requested):
Владислав

много замечаний из передыдущего ревью не отработано - вопросов по ним нет и изменений тоже

нужно или поправить или задать вопросы

Zarecs committed:
add switch

Zarecs commented:
Основные изменения  сделаны в - [Correction of comments]
(https://github.com/Zarecs/spin1/pull/1/commits/4c2e8175c689c4550b923cf5d849d40a986dda2d), а именно: 
 1) Текст ошибок на английском, кроме мест проверяемых тестом
 2) Удалены лишние проверки не используемые в тестах
 3) Удалены лишние комментарии у функций.
 4) удалена  инициализация лишних переменных 
 5) Исправлена оборачиваемость на %w 

в новом коммте - [add switch]
(https://github.com/Zarecs/spin1/pull/1/commits/90b9de4adc871729b9303e6d6b28ef653437601d)
 исправлен блок else if, применил switch

dirtymew reviewed (approved):
Владислав

теперь все отлично исправлено
